# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../libnetfilter_queue/0001-include-sys-types.h.patch
# Copyright (C) 2013 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

include <sys/types.h> to get u_int*_t types

--- ./src/libnetfilter_queue.c.orig	2013-01-01 23:46:25.895771925 +0100
+++ ./src/libnetfilter_queue.c	2013-01-01 23:46:43.784189811 +0100
@@ -28,6 +28,7 @@
 #include <time.h>
 #include <errno.h>
 #include <netinet/in.h>
+#include <sys/types.h>
 #include <sys/socket.h>
 
 #include <libnfnetlink/libnfnetlink.h>
--- ./src/nlmsg.c.orig	2013-01-01 23:47:16.468954700 +0100
+++ ./src/nlmsg.c	2013-01-01 23:47:44.493622220 +0100
@@ -11,6 +11,7 @@
 #include <endian.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sys/types.h>
 
 #include <libmnl/libmnl.h>
 
--- ./src/extra/ipv6.c.orig	2013-01-01 23:48:22.130480381 +0100
+++ ./src/extra/ipv6.c	2013-01-01 23:48:44.326993966 +0100
@@ -14,6 +14,7 @@
 #include <stdbool.h>
 #include <arpa/inet.h>
 #include <netinet/ip6.h>
+#include <sys/types.h>
 
 #include <libnetfilter_queue/libnetfilter_queue.h>
 #include <libnetfilter_queue/libnetfilter_queue_ipv6.h>
--- ./src/extra/ipv4.c.orig	2013-01-02 00:07:01.072416894 +0100
+++ ./src/extra/ipv4.c	2013-01-02 00:07:19.536834217 +0100
@@ -13,6 +13,7 @@
 #include <stdbool.h>
 #include <arpa/inet.h>
 #include <netinet/ip.h>
+#include <sys/types.h>
 
 #include <libnetfilter_queue/libnetfilter_queue.h>
 #include <libnetfilter_queue/libnetfilter_queue_ipv4.h>
--- ./src/extra/udp.c.orig	2013-01-02 00:07:45.613433362 +0100
+++ ./src/extra/udp.c	2013-01-02 00:08:33.374500654 +0100
@@ -15,6 +15,7 @@
 #include <netinet/ip.h>
 #include <netinet/ip6.h>
 #include <netinet/udp.h>
+#include <sys/types.h>
 
 #include <libnetfilter_queue/libnetfilter_queue.h>
 #include <libnetfilter_queue/libnetfilter_queue_udp.h>
